src/DataSource.ts

```diff
@@ -24,6 +26,49 @@ export class GithubDataSource extends DataSourceWithBackend<GitHubQuery, GithubD
     };
   }
 
+  query(request: DataQueryRequest<GitHubQuery>): Observable<DataQueryResponse> {
+    request.targets.forEach((target) => {
+      const issueTimeFieldReference: { [x: number]: string } = {
+        0: 'CreatedAt',
+        1: 'ClosedAt',
+      };
+
+      const pullRequestTimeFieldReference: { [x: number]: string } = {
+        0: 'ClosedAt',
+        1: 'CreatedAt',
+        2: 'MergedAt',
+        3: 'None',
+      };
+
+      const workflowTimeFieldReference: { [x: number]: string } = {
+        0: 'CreatedAt',
+        1: 'UpdatedAt',
+      };
+
+      let properties: Partial<GitHubQuery> = { app: request.app, queryType: target.queryType };
+
+      if (target.queryType === 'Issues') {
+        properties['timeField'] = issueTimeFieldReference[target.options?.timeField ?? 0];
```

Comment:
I like how you don't just toss in the number here however this approach is not error prone. What if someone adds a new enum value to any of the types there? Your code will break.

Instead you can use https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings
like:
`IssueTimeField[target.options?.timeField ?? 0];`

Can you please also extract this reporting logic to its own function?
Any reason you are accessing the timeField property as an index signature?